publish/src/commands/settle.js

```
@@ -104,20 +104,29 @@ const settle = async ({
 
 	const { number: currentBlock } = await web3.eth.getBlock('latest');
 
-	const getContract = ({ label, source }) =>
-		new web3.eth.Contract(
-			getSource({ contract: source }).abi,
-			getTarget({ contract: label }).address
-		);
+	const versions = getVersions({ byContract: true });
+
+	const getContract = ({ label, source = label, blockNumber }) => {
+		let { address } = getTarget({ contract: label });
+
+		if (blockNumber) {
+			// look for the right contract based off the block
+			for (const entry of versions[source]) {
+				// assumes sorted by block number ascending
```

nit: Why make this assumption? If getVersions ever changes and there is not test for this, this will fail silently.

E.g. In the loop, record a distance to the required block, and pick the entry with the minimum (signed) distance.

---

it's a fair nit. Updated.